### PR TITLE
serial number helper

### PR DESCRIPTION
**Pull request by quozl**
On commodity x86 and x86_64 hardware the serial number is available to
root in /sys/class/dmi/id/product_serial, but Sugar does not run as
root, so per commit 612842c the string "Not available" is shown.

Add a polkit helper to read the serial number.

### CODE REVIEW

File: bin/sugar-serial-number-helper [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+device=/sys/class/dmi/id
+
+read value < $device/product_serial
+echo "$value"
```
Comment (samdroid-apps) [inline bin/sugar-serial-number-helper]: Could this use cat an be more simple?  Do we need the device variable?

File: extensions/cpsection/aboutcomputer/model.py [hunk 2]
```diff
@@ -46,11 +48,38 @@ def print_aboutcomputer():
...
+
+    def _helper():
+        binary = 'sugar-serial-number-helper'
+        for path in os.environ['PATH'].split(os.pathsep):
+            binary_path = os.path.join(path, binary)
+            if os.path.exists(binary_path):
+                return binary_path
+        return None
+
+    helper = _helper()
```
Comment (samdroid-apps) [inline extensions/cpsection/aboutcomputer/model.py]: I don't think closures are in favour around here.  Can't it just be a loop with a `break` statement?

File: extensions/cpsection/aboutcomputer/model.py [hunk 3]
```diff
@@ -46,11 +48,38 @@ def print_aboutcomputer():
...
+    if helper is None:
+        return _not_available
+
+    cmd = 'pkexec %s' % helper
```
Comment (samdroid-apps) [inline extensions/cpsection/aboutcomputer/model.py]: pkexec seems to automatically resolve the full path of the executable.  Maybe it is better to let pkexec do it instead of doing the exact same thing ourselves except probably not dealing with as many edge cases?

Comment (samdroid-apps) [inline extensions/cpsection/aboutcomputer/model.py]: eg. to test run `pkexec sugar-backlight-helper --get-max-brightness`.  Works fine for me (fc24 rawhide).

**Comment by samdroid-apps** [comment]
Looks great, just a few little things.

File: bin/sugar-serial-number-helper (hunk 1 above)
Comment (quozl) [inline bin/sugar-serial-number-helper]: Using ```cat``` would add delay while another binary is started as a subprocess of the shell.  ```read``` is a built-in for the shell.  The device variable follows the style of ```sugar-backlight-helper```.

File: extensions/cpsection/aboutcomputer/model.py (hunk 2 above)
Comment (quozl) [inline extensions/cpsection/aboutcomputer/model.py]: Again, this copied the style set by the backlight helper.

File: extensions/cpsection/aboutcomputer/model.py (hunk 3 above)
Comment (quozl) [inline extensions/cpsection/aboutcomputer/model.py]: Good idea, but again, I followed the existing style.  Would you like to patch the backlight helper?

**Comment by quozl** [comment]
Thanks for the review.  Please make it clearer which little things are questions for clarification, and which you think need to be fixed before merge.
